tests/test_indexer.py

```diff
@@ -194,29 +195,73 @@ def test_indexed_file_access_error(self):
                                                          files=files,
                                                          excluded_files=[inaccesssible_filename.replace(".", "_")])
 
+    def test_notify(self):
+        def _notify(subscription, bundle_id="i.v"):
+            notify(subscription_id=subscription["id"], subscription=subscription, bundle_id=bundle_id, logger=logger)
+        with self.assertRaisesRegex(requests.exceptions.InvalidURL, "Invalid URL 'http://': No host supplied"):
+            _notify(subscription=dict(id="", es_query={}, callback_url="http://"))
+        with self.assertRaisesRegex(AssertionError, "Unexpected scheme for callback URL"):
+            _notify(subscription=dict(id="", es_query={}, callback_url=""))
+        with self.assertRaisesRegex(AssertionError, "Unexpected scheme for callback URL"):
+            _notify(subscription=dict(id="", es_query={}, callback_url="wss://localhost"))
+        try:
+            environ_backup = os.environ
+            os.environ = dict(DSS_DEPLOYMENT_STAGE="prod")
+            with self.assertRaisesRegex(AssertionError, "Unexpected scheme for callback URL"):
+                _notify(subscription=dict(id="", es_query={}, callback_url="http://example.com"))
+            with self.assertRaisesRegex(AssertionError, "Callback hostname resolves to forbidden network"):
+                _notify(subscription=dict(id="", es_query={}, callback_url="https://localhost"))
+        finally:
+            os.environ = environ_backup
+
+    def delete_subscription(self, subscription_id):
+        self.assertDeleteResponse(
+            str(UrlBuilder().set(path=f"/v1/subscriptions/{subscription_id}").add_query("replica", self.replica)),
+            requests.codes.ok,
+            headers=self.get_auth_header()
+        )
+
     def test_subscription_notification_successful(self):
```

Comment:
`test_subscription_notification_successful` still has a fair amount of internal duplication.  and now that you've collapsed it into one method, it should be even easier to dedup.

```
    for verify_payloads, subscribe_kwargs in ((True, dict(hmac_secret_key=PostTestHandler.hmac_secret_key)),
                                              (False, dict()):
        PostTestHandler.verify_payloads = verify_payloads
        bundle_key = self.load_test_data_bundle_for_path("fixtures/smartseq2/paired_ends")
        sample_event = self.create_sample_bundle_created_event(bundle_key)
        self.process_new_indexable_object(sample_event, logger)

        ElasticsearchClient.get(logger).indices.create(self.subscription_index_name)
        subscription_id = self.subscribe_for_notification(smartseq2_paired_ends_query,
                                                          f"http://{HTTPInfo.address}:{HTTPInfo.port}",
                                                          **subscribe_kwargs)

        bundle_key = self.load_test_data_bundle_for_path("fixtures/smartseq2/paired_ends")
        sample_event = self.create_sample_bundle_created_event(bundle_key)
        self.process_new_indexable_object(sample_event, logger)
        prefix, _, bundle_id = bundle_key.partition("/")
        self.verify_notification(subscription_id, smartseq2_paired_ends_query, bundle_id)
```

Reply:
Deduplicated code per your suggestion.